contributors/design-proposals/statefulset-update.md

```diff
@@ -0,0 +1,846 @@
+1. If the number of PodTemplates in the StatefulSet's revision history is 
+greater than the StatefulSet's `.Spec.RevisionHistoryLimit`, the 
+StatefulSet controller will delete PodTemplates, starting with the head of 
+the revision history, until the size of the revision history is equal to 
+the StatefulSet's `.Spec.RevisionHistoryLimit`.
+
+### Template Updates
+The StatefulSet controller will create PodTemplates upon mutation of the 
+`.Spec.Template` of a StatefulSet.
+
+1. When the StafefulSet controller observes a mutation to a StatefulSet's 
+ `.Spec.Template` it will compare the `.Spec.TemplateGeneration` to the 
+ `.Status.UpdatedTemplateGeneration`.
+1. If the `.Spec.TemplateGeneration` is equivalent to the 
+`.Status.UpdatedTemplateGeneration`, no update has occurred. Note that, in the 
+event that both are nil, they are considered to be equivalent, and we expect 
+this to occur after an initial upgrade to a version of Kubernetes that supports 
+StatefulSet update form one that does not.
+1. If the `.Status.TemplateGeneration` field is nil, and the 
+`.Spec.TemplateGeneration` is not nil, then the StatefulSet has no revision 
+history. To initialize its revision history, the StatefulSet controller will 
+set both `.Status.CurrentTemplateGeneration` and `.Status.UpdatedTemplateGeneration` 
+to `.Spec.TemplateGeneration` and 
+[create a new PodTemplate](#podtemplate-creation). 
+1. If the `.Status.CurrentTemplateGeneration` is not nil, and if the 
+`.Spec.TemplateGeneration` is not equal to the `.Status.UpdatedTemplateGeneration`, 
+the StatefulSet controller will do the following.
+    1. The controller will 
+    [reconstruct the revision history](#history-reconsturction) of the 
+    StatefulSet.
+    1. If the revision history of the StatefulSet contains any PodTemplate 
+    whose `.PodTemplateSpec` is semantically, deeply equivalent to the 
+    StatefulSet's `.Spec.Template`, the controller will record the revision of
+    all such templates.
+    1. The controller will update the `StatefulSetPodTemplateGeneration` label 
+    of all Pods that match any revision recorded above.
+    1. The controller will update the StatefulSet's 
+    `.Status.UpdatedTemplateGeneration` to the new revision.
+
+### PodTemplate Selection
+When the StatefulSet controller creates the Pods in a StatefulSet, it will use 
+the following criteria to select the PodTemplate used to create a Pod. These 
+criteria allow the controller to continue to make progress toward its target 
+state, while respecting its guarantees and allowing for rolling updates back 
+and forward.
+
+1. If the StatefulSet's `.Spec.TemplateGeneration` is nil, then the cluster 
+has been upgraded from a version that does not support StatefulSet update to 
+a version that does. 
+    1. In this case the `.Spec.Template` is the current revision, 
+    and no Pods in the StatefulSet should be labeled with a 
+    `StatefulSetPodTemplateGeneration` label. 
```

Comment:
Maybe I missed it. Is "labeling pods with a `StatefulSetPodTemplateGeneration` label" (when creating a StatefulSet pod) mentioned in this proposal?